docs/ListIterator.md

```diff
@@ -0,0 +1,250 @@
+
+{% raw %}
+```tsx
+import { ListBase, ListIterator } from 'react-admin';
+import { OrderedList, ListItem } from 'my-favorite-ui-lib';
+import { useQuery } from '@tanstack/react-query';
+import { fetchPostAnalytics } from './fetchPostAnalytics';
+
+const DashboardMostVisitedPosts = () => {
+    const { data, isPending } = useQuery({
+        queryKey: ['dashboard', 'posts'],
+        queryFn: fetchPostAnalytics
+    });
+
+    return (
+        <OrderedList>
+            <ListIterator
+                isPending={isPending}
+                render={record => <ListItem>{record.title} - {record.views}</ListItem>}
+            />
+        </OrderedList>
+    );
+}
+```
+{% endraw %}
+
+
+## `pending`
+
+To provide a custom UI while the data is loading use the `pending` prop.
+
+{% raw %}
+```jsx
+import { ListBase, ListIterator } from 'react-admin';
+import { OrderedList, ListItem, Skeleton } from 'my-favorite-ui-lib';
+
+const DashboardMostVisitedPosts = () => (
+    <ListBase resource="posts" sort={{ field: 'views', order: 'DESC' }} page={1} perPage={20}>
+        <OrderedList>
+            <ListIterator
+                pending={<Skeleton />}
+                render={record => <ListItem>{record.title} - {record.views}</ListItem>}
+            />
+        </OrderedList>
+    </ListBase>
+);
+```
+{% endraw %}
+
+## `render`
+
+If provided, `ListIterator` will call the `render` prop for each record. This is useful when the components you render don't leverage the [`RecordContext`](./useRecordContext.md):
+
+{% raw %}
+```tsx
+import { ListBase, ListIterator } from 'react-admin';
+import { OrderedList, ListItem } from 'my-favorite-ui-lib';
+
+const DashboardMostVisitedPosts = () => (
+    <ListBase resource="posts" sort={{ field: 'views', order: 'DESC' }} page={1} perPage={20}>
+        <OrderedList>
+            <ListIterator
+                render={record => <ListItem>{record.title} - {record.views}</ListItem>}
+            />
+        </OrderedList>
+    </ListBase>
+);
+```
+{% endraw %}
+
+**Note**: You can't provide both the `children` and the `render` props. If both are provided, `<ListIterator>` will use the `render` prop.
+
+## `resource`
+
+Although `<ListIterator>` reads the resource from the closest [`<ResourceContext>`](./Resource.md#resource-context), you may provide it yourself when no such context is available (e.g. in a [dashboard](./Admin.md#dashboard) or a [custom page](./Admin.md#adding-custom-pages)):
+
+{% raw %}
+```jsx
+import { ListBase, ListIterator } from 'react-admin';
+import { OrderedList, ListItem } from 'my-favorite-ui-lib';
+
+const DashboardMostVisitedPosts = () => (
+    <ListBase resource="posts" sort={{ field: 'views', order: 'DESC' }} page={1} perPage={20}>
+        <OrderedList>
+            <ListIterator
+                render={record => <ListItem>{record.title} - {record.views}</ListItem>}
+            />
```

Comment:
you forgot to use `resource` in this example.

Also, it's not obvious why `<ListIterator>` would require a `resource` to operate. Provided you have a `data` array, it should be all you need to iterate over it, correct?

EDIT: as I thought, it's actually never used. You can remove this prop altogether I believe.